src/markdown_exec/mkdocs_plugin.py

```diff
@@ -128,6 +129,8 @@ def on_post_build(self, *, config: MkDocsConfig) -> None:  # noqa: ARG002,D102
         else:
             os.environ["MKDOCS_CONFIG_DIR"] = self.mkdocs_config_dir
 
+        _shutdown_kernels()
```

Comment:
issue: `on_post_build` runs after *each* build during live-reload, so every time the site is rebuilt/reloaded. Maybe shutting down kernels is not something we'd want to do here, but instead in [`on_shutdown`](https://www.mkdocs.org/dev-guide/plugins/#on_shutdown)? Clearing sessions should definitely happen in `on_post_build` though, not sure if that's only what `_shutdown_kernels` does :slightly_smiling_face: